### PR TITLE
Add missing CLI and INI options for a consistent API

### PR DESCRIPTION
**Pull request by astrofrog**
It would be nice to be able to customize the default tolerance and style to use globally as both have defaults set by 'historical reasons' that aren't necessarily sensible.

For style, one of the issues is that actually style could be more than just a string - e.g. a dict and so on. Is there a way to set global options such as these in e.g. conftest.py? (if so I can document that too).

Remaining TODOs (if we go ahead with this):

* [x] Add some documentation about the options
* [x] Add tests

### CODE REVIEW

**Comment by Cadair** [comment]
> Is there a way to set global options such as these in e.g. conftest.py?

You can set these options in the pytest config (normally setup.cfg) but I don't know how that interacts with richer python objects? maybe we would need to parse a string in to a python object?

**Comment by ConorMacBride** [comment]
> > Is there a way to set global options such as these in e.g. conftest.py?
> 
> You can set these options in the pytest config (normally setup.cfg) but I don't know how that interacts with richer python objects? maybe we would need to parse a string in to a python object?

The pytest global options (probably) only supports returning [one of these types](https://docs.pytest.org/en/7.1.x/reference/reference.html?highlight=addoption#pytest.Parser.addini). So I think it would be best to just recommend adding custom rc params to a new style file within the test directory, whose path (or URL) can just be referred to in the global pytest option as a string.

File: pytest_mpl/plugin.py [hunk 1]
```diff
@@ -196,6 +204,12 @@ def pytest_configure(config):
         if results_dir is not None:
             results_dir = os.path.abspath(results_dir)
 
+        default_style = (config.getoption("--mpl-default-style") or
+                         config.getini("mpl-default-style"))
```
Comment (ConorMacBride) [inline pytest_mpl/plugin.py:208]: The default would need to be moved from the decorator to here:
```suggestion
                         config.getini("mpl-default-style") or
                         "classic")
```

File: pytest_mpl/plugin.py [hunk 2]
```diff
@@ -196,6 +204,12 @@ def pytest_configure(config):
...
+
+        default_tolerance = (config.getoption("--mpl-default-tolerance") or
+                             config.getini("mpl-default-tolerance"))
```
Comment (ConorMacBride) [inline pytest_mpl/plugin.py:211]: The options will return strings so need to cast to float:
```suggestion
        default_tolerance = float(config.getoption("--mpl-default-tolerance") or
                                  config.getini("mpl-default-tolerance") or
                                  "2")
```

**Review by ConorMacBride — COMMENTED** [review]
These changes should get the tests passing. I think we should do a new major version soon with the default style changed from `"classic"` to `"default"`, as I think any changes to the Matplotlib default style are outside the scope of this package and should be made apparent by default when testing on different Matplotlib version.

**Comment by ConorMacBride** [comment]
Hi @astrofrog, there are some other options (noted in #198) which could be added to make the config more consistent. Do you agree we should add these options?

- [ ] Add `mpl-baseline-path` ini option (relative to where pytest was run).
- [ ] Add `--mpl-use-full-test-name` CLI option.
- [ ] Add `mpl-hash-library` ini option (relative to where pytest was run).
- [x] Add `--mpl-default-tolerance` CLI and `mpl-default-tolerance` ini options.
- [x] Add `--mpl-default-style` CLI and `mpl-default-style` ini options.
- [ ] Add `--mpl-default-backend` CLI and `mpl-default-backend` ini options.
- [ ] Add `mpl-generate-summary` ini option.

If so, would you like me to open a separate PR, or shall I just add them to this PR?

**Comment by astrofrog** [comment]
Sounds good - feel free to just add them to this PR if you like!

**Review by astrofrog — COMMENTED** [review]
Thanks for all the work on this!

**Pull request merged**